### ISSUE TITLE
No decompiled source function with name (RabbitMq.Handler.handle) at org.elixir_lang.beam.psi.impl.ModuleImpl.setMirror(ModuleImpl.kt:70)

### ISSUE THREAD

**Issue by alexxero** [KronicDeth/intellij-elixir#2386]
# System

Plugin Version: 12.0.1
Application: RubyMine (2021.3)
Operating System: Mac OS X (11.6)

# Event

### Stacktrace

```
java.lang.Throwable: No decompiled source function with name (RabbitMq.Handler.handle)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:182)
	at org.elixir_lang.beam.psi.impl.ModuleImpl.setMirror(ModuleImpl.kt:70)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.setMirror(ModuleElementImpl.java:55)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.setMirrors(ModuleElementImpl.java:46)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.setMirrors(ModuleElementImpl.java:37)
	at org.elixir_lang.beam.psi.BeamFileImpl.setMirror(BeamFileImpl.kt:411)
	at org.elixir_lang.beam.psi.BeamFileImpl.getMirror$lambda-6$lambda-5(BeamFileImpl.kt:392)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeNonCancelableSection$3(CoreProgressManager.java:222)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:698)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:646)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:237)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:221)
	at org.elixir_lang.beam.psi.BeamFileImpl.getMirror(BeamFileImpl.kt:391)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.getMirror(ModuleElementImpl.java:115)
	at org.elixir_lang.beam.psi.impl.ModuleImpl.getNavigationElement(ModuleImpl.kt:111)
	at org.elixir_lang.psi.impl.QualifiableAliasImplKt.toModulars(QualifiableAliasImpl.kt:111)
	at org.elixir_lang.psi.impl.QualifiableAliasImplKt.toModulars(QualifiableAliasImpl.kt:96)
	at org.elixir_lang.psi.impl.QualifiableAliasImplKt.maybeModularNameToModulars(QualifiableAliasImpl.kt:80)
	at org.elixir_lang.psi.impl.PsiElementImplKt.maybeModularNameToModulars(PsiElementImpl.kt:181)
	at org.elixir_lang.psi.impl.call.qualification.QualifiedImplKt.qualifiedToModulars(QualifiedImpl.kt:19)
	at org.elixir_lang.reference.resolver.Callable.resolveQualified(Callable.kt:142)
	at org.elixir_lang.reference.resolver.Callable.resolveInScope(Callable.kt:98)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:88)
	at org.elixir_lang.reference.resolver.Callable.resolveAll(Callable.kt:84)
	at org.elixir_lang.reference.resolver.Callable.resolvePreferred(Callable.kt:74)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:39)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:35)
	at org.elixir_lang.reference.resolver.Callable.resolve(Callable.kt:30)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$resolve$1(ResolveCache.java:154)
	at com.intellij.openapi.util.Computable.get(Computable.java:18)
	at com.intellij.psi.impl.source.resolve.ResolveCache.lambda$loggingResolver$4(ResolveCache.java:260)
	at com.intellij.openapi.util.Computable.get(Computable.java:18)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:238)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolve(ResolveCache.java:154)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:169)
	at com.intellij.psi.impl.source.resolve.ResolveCache.resolveWithCaching(ResolveCache.java:162)
	at org.elixir_lang.reference.ResolveWithCachingComputable.compute(Callable.kt:593)
	at org.elixir_lang.reference.ResolveWithCachingComputable.compute(Callable.kt:585)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:852)
	at org.elixir_lang.reference.CallableKt.resolveWithCaching(Callable.kt:583)
	at org.elixir_lang.reference.CallableKt.access$resolveWithCaching(Callable.kt:1)
	at org.elixir_lang.reference.Callable.multiResolve(Callable.kt:105)
	at org.elixir_lang.ReferenceKt.safeMultiResolve(reference.kt:16)
	at org.elixir_lang.annotator.Callable$annotate$1.visitPlainCall(Callable.kt:97)
	at org.elixir_lang.annotator.Callable$annotate$1.visitNonModuleAttributeCall(Callable.kt:87)
	at org.elixir_lang.annotator.Callable$annotate$1.visitCall(Callable.kt:47)
	at org.elixir_lang.annotator.Callable$annotate$1.visitElement(Callable.kt:78)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:273)
	at org.elixir_lang.psi.impl.ElixirMatchedQualifiedParenthesesCallImpl.accept(ElixirMatchedQualifiedParenthesesCallImpl.java:38)
	at org.elixir_lang.annotator.Callable.annotate(Callable.kt:39)
```

**Comment by KronicDeth** [comment]
Can you take the .beam file, put it in a zip file and add it as a file attachment to this issue?  Then I can use it in a regression test.

**Comment by alexxero** [comment]
Absolutely!
Here it is.
I hope that the needed file.
Let me know if not.
[Elixir.RabbitMq.Worker.beam.zip](https://github.com/KronicDeth/intellij-elixir/files/7694545/Elixir.RabbitMq.Worker.beam.zip)

**Comment by pjar** [comment]
Started to happen to me too, out of the blue. Was working fine for several hours then it suddenly froze the UI and now it keeps doing it when I restart RubyMine.  I even uninstalled it RM few times, cleared the cache folder, but it keeps freezing the UI no matter what.

**Comment by KronicDeth** [comment]
@pjar if it is freezing the UI then it is likely a large decompiled file.  There are heuristics to detect this because the Unicode files in Elixir stdlib had this issue before.

UI freezes have special logs:

1.  Help > Show Logs in Finder
2. Zip any folders named `threadDumps-freeze-*`
3. Attach the zips to a comment here.

**Comment by KronicDeth** [comment]
I'm super confused.  The `Elixir.RabbitMq.Worker.beam` doesn't have a `handle` function in the `DbgI` or `ExpT` chunks either.  I don't know why the mirror code was trying to find one.

**Comment by KronicDeth** [comment]
*face palm*  I didn't even notice you gave me a different `.beam` file than the error.

I need the `Elixir.RabbitMq.Handler.beam` @alexxero.  You gave me the `Elixir.RabbitMq.Worker.beam`.

**Comment by KronicDeth** [comment]
I found the `rabbit_mq` package (https://hex.pm/packages/rabbit_mq), but it doesn't have a `Elixir.RabbitMq.Handler.beam`.

**Comment by alexxero** [comment]
[Elixir.RabbitMq.Handler.beam.zip](https://github.com/KronicDeth/intellij-elixir/files/7810550/Elixir.RabbitMq.Handler.beam.zip)
That should be the file

**Comment by KronicDeth** [comment]
It's got an external function capture like `Ecto.Changeset.validate_number`

![Screen Shot 2022-01-05 at 9 49 27 AM](https://user-images.githubusercontent.com/298259/148247024-24038dfe-a379-47a8-a4f1-4de83d039f32.png)

So, it's fixed by #2441 also.

![Screen Shot 2022-01-05 at 9 49 27 AM](https://user-images.githubusercontent.com/298259/148248924-7dad60ef-9faf-4704-8f08-adc19a52e755.png)
